### DISCUSSION TITLE
Traffic & File Ticker in Web UI

### DISCUSSION THREAD

**Discussion by camm9909** [slskd/slskd#547] [Ideas]
Would be nice to have a total traffic and file ticker within the 'shares' option or somewhere else in the UI.
I'm currently using NetHogs for this but would be nice to have it in the WebUI :)

![image](https://user-images.githubusercontent.com/73704121/179019754-f2d0268e-1f65-48d6-a967-76813737afe9.png)

**Comment by jpdillingham** [comment]
Thanks for the idea!

I'm currently thinking about the best way to tackle the transfer history problem, as it is probably the biggest shortcoming of the app right now.  I personally need it so I can start getting a better idea of why transfers fail, and it is needed to support downloads that persist across application restarts.

What you suggest is another use case for this history; once I start keeping track of transfers these stats are trivial to compute.

**Reply by jpdillingham** [reply]
This is tracked in #387, which will enable these statistics.  I'll need to tweak that a little so that I track _all_ transfers, but that's a pretty small tweak under the hood.

**Reply by camm9909** [reply]
I suppose there are a number of ways you could implement this, paged (per day or per week) logs with compression to reduce storage overhead. Or have user defined macro rotation (ie per year). Or implement a sort of SQL database. I didn't see this issue either so thanks for pointing that out!

**Reply by jpdillingham** [reply]
Yep, persistent data is stored in SQLite databases.  I'm working on implementing storage now so at least historical data will be available later when I implement historicals.
